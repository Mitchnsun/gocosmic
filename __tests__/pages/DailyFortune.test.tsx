import DailyFortune from '@/app/[locale]/applications/daily-fortune/page';

import { render } from '../test-utils';

describe('DailyFortune Page', () => {
  it('should render the daily fortune page correctly', () => {
    const { container } = render(<DailyFortune />);

    expect(container).toMatchSnapshot();
  });
});
