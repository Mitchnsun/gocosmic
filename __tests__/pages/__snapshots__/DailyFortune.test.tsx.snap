// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DailyFortune Page > should render the daily fortune page correctly 1`] = `
<div>
  <div
    class="text-ghost relative pt-10"
  >
    <main
      class="m-auto flex max-w-7xl flex-col items-center gap-10 px-4 pb-4"
    >
      <div
        class="flex flex-col items-center justify-between gap-8 lg:flex-row"
      >
        <img
          alt="Daily Fortune Logo"
          class="overflow-hidden rounded-2xl"
          data-nimg="1"
          decoding="async"
          height="100"
          loading="lazy"
          src="/_next/image?url=%2Fprojects%2Fdaily-fortune%2Fapp-icon.png&w=256&q=75"
          srcset="/_next/image?url=%2Fprojects%2Fdaily-fortune%2Fapp-icon.png&w=128&q=75 1x, /_next/image?url=%2Fprojects%2Fdaily-fortune%2Fapp-icon.png&w=256&q=75 2x"
          style="color: transparent;"
          width="100"
        />
        <div
          class="text-center"
        >
          <h1
            class="mb-4 text-2xl font-extrabold sm:text-4xl"
          >
            Daily Fortune
          </h1>
          <p
            class="text-lg text-gray-400"
          >
            Your daily dose of inspiration and cosmic wisdom
          </p>
        </div>
      </div>
      <section
        aria-labelledby="overview-heading"
        class="w-full"
      >
        <div
          class="relative rounded-lg bg-gray-800 px-6 py-8 lg:p-8"
        >
          <div
            class="mb-6 flex items-center gap-4"
          >
            <svg
              aria-hidden="true"
              class="h-8 w-8 shrink-0 text-purple-400"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.75 12.75a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM7.5 15.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM8.25 17.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM9.75 15.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM10.5 17.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM12 15.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM12.75 17.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM14.25 15.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM15 17.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM16.5 15.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM15 12.75a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM16.5 13.5a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Z"
              />
              <path
                clip-rule="evenodd"
                d="M6.75 2.25A.75.75 0 0 1 7.5 3v1.5h9V3A.75.75 0 0 1 18 3v1.5h.75a3 3 0 0 1 3 3v11.25a3 3 0 0 1-3 3H5.25a3 3 0 0 1-3-3V7.5a3 3 0 0 1 3-3H6V3a.75.75 0 0 1 .75-.75Zm13.5 9a1.5 1.5 0 0 0-1.5-1.5H5.25a1.5 1.5 0 0 0-1.5 1.5v7.5a1.5 1.5 0 0 0 1.5 1.5h13.5a1.5 1.5 0 0 0 1.5-1.5v-7.5Z"
                fill-rule="evenodd"
              />
            </svg>
            <h2
              class="text-xl font-bold sm:text-2xl lg:text-3xl"
              id="overview-heading"
            >
              About Daily Fortune
            </h2>
          </div>
          <div
            class="space-y-4 text-gray-300"
          >
            <p
              class="text-lg"
            >
              Daily Fortune is a delightful mobile application that brings a touch of cosmic magic to your routine. Each day, receive a personalized fortune to brighten your day.
            </p>
            <p>
              Whether you're looking for a morning boost, a phrase to keep you on track, or a message to share, Daily Fortune fits into your routine to spread positivity and clarity.
            </p>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="features-heading"
        class="w-full"
      >
        <div
          class="rounded-lg bg-gray-800 px-6 py-8 lg:p-8"
        >
          <div
            class="mb-6 flex items-center gap-4"
          >
            <svg
              aria-hidden="true"
              class="h-8 w-8 shrink-0 text-yellow-400"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M9 4.5a.75.75 0 0 1 .721.544l.813 2.846a3.75 3.75 0 0 0 2.576 2.576l2.846.813a.75.75 0 0 1 0 1.442l-2.846.813a3.75 3.75 0 0 0-2.576 2.576l-.813 2.846a.75.75 0 0 1-1.442 0l-.813-2.846a3.75 3.75 0 0 0-2.576-2.576l-2.846-.813a.75.75 0 0 1 0-1.442l2.846-.813A3.75 3.75 0 0 0 7.466 7.89l.813-2.846A.75.75 0 0 1 9 4.5ZM18 1.5a.75.75 0 0 1 .728.568l.258 1.036c.236.94.97 1.674 1.91 1.91l1.036.258a.75.75 0 0 1 0 1.456l-1.036.258c-.94.236-1.674.97-1.91 1.91l-.258 1.036a.75.75 0 0 1-1.456 0l-.258-1.036a2.625 2.625 0 0 0-1.91-1.91l-1.036-.258a.75.75 0 0 1 0-1.456l1.036-.258a2.625 2.625 0 0 0 1.91-1.91l.258-1.036A.75.75 0 0 1 18 1.5ZM16.5 15a.75.75 0 0 1 .712.513l.394 1.183c.15.447.5.799.948.948l1.183.395a.75.75 0 0 1 0 1.422l-1.183.395c-.447.15-.799.5-.948.948l-.395 1.183a.75.75 0 0 1-1.422 0l-.395-1.183a1.5 1.5 0 0 0-.948-.948l-1.183-.395a.75.75 0 0 1 0-1.422l1.183-.395c.447-.15.799-.5.948-.948l.395-1.183A.75.75 0 0 1 16.5 15Z"
                fill-rule="evenodd"
              />
            </svg>
            <h2
              class="text-xl font-bold sm:text-2xl lg:text-3xl"
              id="features-heading"
            >
              Key Features
            </h2>
          </div>
          <div
            class="space-y-4 text-gray-300"
          >
            <ul
              class="grid gap-3 md:grid-cols-2"
            >
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-yellow-400"
                />
                <span>
                  New cosmic fortunes delivered every day
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-yellow-500"
                />
                <span>
                  A chosen message to brighten your day: fortune cookie, motivational quote, or positive affirmation
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-yellow-600"
                />
                <span>
                  Clean, responsive interface with smooth animations
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-yellow-700"
                />
                <span>
                  Space-inspired design that captures the wonder of the universe
                </span>
              </li>
            </ul>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="technology-heading"
        class="w-full"
      >
        <div
          class="rounded-lg bg-gray-800 px-6 py-8 lg:p-8"
        >
          <div
            class="mb-6 flex items-center gap-4"
          >
            <svg
              aria-hidden="true"
              class="h-8 w-8 shrink-0 text-blue-400"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.447 3.026a.75.75 0 0 1 .527.921l-4.5 16.5a.75.75 0 0 1-1.448-.394l4.5-16.5a.75.75 0 0 1 .921-.527ZM16.72 6.22a.75.75 0 0 1 1.06 0l5.25 5.25a.75.75 0 0 1 0 1.06l-5.25 5.25a.75.75 0 1 1-1.06-1.06L21.44 12l-4.72-4.72a.75.75 0 0 1 0-1.06Zm-9.44 0a.75.75 0 0 1 0 1.06L2.56 12l4.72 4.72a.75.75 0 0 1-1.06 1.06L.97 12.53a.75.75 0 0 1 0-1.06l5.25-5.25a.75.75 0 0 1 1.06 0Z"
                fill-rule="evenodd"
              />
            </svg>
            <h2
              class="text-xl font-bold sm:text-2xl lg:text-3xl"
              id="technology-heading"
            >
              Technology Stack
            </h2>
          </div>
          <div
            class="space-y-4 text-gray-300"
          >
            <p
              class="text-lg"
            >
              Built with modern web technologies to deliver a fast, beautiful, and reliable experience:
            </p>
            <ul
              class="grid gap-3 md:grid-cols-2"
            >
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-blue-400"
                />
                <span>
                  Next.js - React framework for production-ready applications
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-blue-500"
                />
                <span>
                  TypeScript - Type-safe development for reliability
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-blue-600"
                />
                <span>
                  Tailwind CSS - Utility-first styling for cosmic aesthetics
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-blue-700"
                />
                <span>
                  React Native - Native mobile experience for iOS and Android
                </span>
              </li>
            </ul>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="ai-heading"
        class="w-full"
      >
        <div
          class="rounded-lg bg-gray-800 px-6 py-8 lg:p-8"
        >
          <div
            class="mb-6 flex items-center gap-4"
          >
            <svg
              aria-hidden="true"
              class="h-8 w-8 shrink-0 text-green-400"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.5 7.5h-9v9h9v-9Z"
              />
              <path
                clip-rule="evenodd"
                d="M8.25 2.25A.75.75 0 0 1 9 3v.75h2.25V3a.75.75 0 0 1 1.5 0v.75H15V3a.75.75 0 0 1 1.5 0v.75h.75a3 3 0 0 1 3 3v.75H21A.75.75 0 0 1 21 9h-.75v2.25H21a.75.75 0 0 1 0 1.5h-.75V15H21a.75.75 0 0 1 0 1.5h-.75v.75a3 3 0 0 1-3 3h-.75V21a.75.75 0 0 1-1.5 0v-.75h-2.25V21a.75.75 0 0 1-1.5 0v-.75H9V21a.75.75 0 0 1-1.5 0v-.75h-.75a3 3 0 0 1-3-3v-.75H3A.75.75 0 0 1 3 15h.75v-2.25H3a.75.75 0 0 1 0-1.5h.75V9H3a.75.75 0 0 1 0-1.5h.75v-.75a3 3 0 0 1 3-3h.75V3a.75.75 0 0 1 .75-.75ZM6 6.75A.75.75 0 0 1 6.75 6h10.5a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75H6.75a.75.75 0 0 1-.75-.75V6.75Z"
                fill-rule="evenodd"
              />
            </svg>
            <h2
              class="text-xl font-bold sm:text-2xl lg:text-3xl"
              id="ai-heading"
            >
              Powered by Intelligence
            </h2>
          </div>
          <div
            class="space-y-4 text-gray-300"
          >
            <p
              class="text-lg"
            >
              Daily Fortune leverages modern AI capabilities to generate unique, engaging, and contextually relevant fortunes that resonate with everyone's daily experiences and aspirations.
            </p>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="cta-heading"
        class="flex w-full flex-col items-center justify-center gap-8 lg:flex-row"
      >
        <img
          alt="Daily Fortune Illustration"
          class="overflow-hidden rounded-lg"
          data-nimg="1"
          decoding="async"
          height="250"
          loading="lazy"
          src="/_next/image?url=%2Fprojects%2Fdaily-fortune%2Fsplashscreen.png&w=640&q=75"
          srcset="/_next/image?url=%2Fprojects%2Fdaily-fortune%2Fsplashscreen.png&w=256&q=75 1x, /_next/image?url=%2Fprojects%2Fdaily-fortune%2Fsplashscreen.png&w=640&q=75 2x"
          style="color: transparent;"
          width="250"
        />
        <div
          class="flex max-w-5xl flex-col items-center gap-6 rounded-lg bg-gray-800 px-6 py-12 lg:p-12"
        >
          <h2
            class="text-center text-3xl font-bold lg:text-4xl"
            id="cta-heading"
          >
            Explore Daily Fortune
          </h2>
          <p
            class="text-center text-lg text-gray-400"
          >
            Discover your cosmic fortune and let the stars guide your journey.
          </p>
          <a
            aria-label="Visit Daily Fortune repository (opens in a new tab)"
            class="bg-royal hover:bg-royal/90 inline-flex items-center gap-2 rounded-full px-6 py-2 text-base font-normal text-neutral-50 transition-colors"
            href="https://github.com/Mitchnsun/daily-fortune"
            rel="noopener noreferrer"
            target="_blank"
          >
            Visit the repository
            <svg
              aria-hidden="true"
              class="h-5 w-5"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M15.75 2.25H21a.75.75 0 0 1 .75.75v5.25a.75.75 0 0 1-1.5 0V4.81L8.03 17.03a.75.75 0 0 1-1.06-1.06L19.19 3.75h-3.44a.75.75 0 0 1 0-1.5Zm-10.5 4.5a1.5 1.5 0 0 0-1.5 1.5v10.5a1.5 1.5 0 0 0 1.5 1.5h10.5a1.5 1.5 0 0 0 1.5-1.5V10.5a.75.75 0 0 1 1.5 0v8.25a3 3 0 0 1-3 3H5.25a3 3 0 0 1-3-3V8.25a3 3 0 0 1 3-3h8.25a.75.75 0 0 1 0 1.5H5.25Z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </section>
    </main>
  </div>
</div>
`;
